src/uct/api/uct.h

```diff
@@ -259,6 +272,26 @@ typedef struct uct_component_attr {
      */
     uct_md_resource_desc_t *md_resources;
 
+    /** Number of connection manager resources */
+    unsigned               cm_resource_count;
+
+    /**
+     * Array holding the connection manager resource.
+     * When used, it should be initialized prior to calling
+     * @ref uct_component_query with a pointer to an array, which is large
+     * enough to hold one connection manager resource entrie.
+     * After the call, this array will be filled with information about the
+     * existing connection manager resource.
+     * In order to allocate this array, you can call @ref uct_component_query
+     * twice: The first time would only obtain the amount of entries required,
+     * i.e. tell you if there is a connection manager availbale on this component,
+     * by specifying @ref UCT_COMPONENT_ATTR_FIELD_CM_RESOURCE_COUNT in
+     * field_mask. Then the array could be allocated with the returned number of
+     * entries, and passed to a second call to @ref uct_component_query, this
+     * time setting field_mask to @ref UCT_COMPONENT_ATTR_FIELD_CM_RESOURCE.
+     */
+    uct_cm_resource_desc_t *cm_resource;
```

Comment:
IMHO we don't need it at all until it doesn't contain any useful data. CM name and component name should be identical. Capability to open CM can be detected by uct_component_query which would return some capability flag.